### PR TITLE
Adding changelog for release v0.10.0

### PR DESCRIPTION
**Pull request by pokearu**
*Description of changes:*
Changes for release `v0.10.0`

*Testing (if applicable):*
`make serve`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/reference/changelog.md [hunk 1]
```diff
@@ -6,10 +6,19 @@ weight: 35
 
 ## Unreleased
```
Comment (chrisnegus) [inline docs/content/en/docs/reference/changelog.md:7]: Can we delete "## Unreleased" ? I think it showed up before and we got rid of it.

File: docs/content/en/docs/reference/changelog.md [hunk 2]
```diff
@@ -6,10 +6,19 @@ weight: 35
...
 
+## [v0.10.0](https://github.com/aws/eks-anywhere/releases/tag/v0.10.0)
+
 ### Added
+- Added support to EKS Anywhere for bare metal cluster provisioning with provider tinkerbell.
```
Comment (chrisnegus) [inline docs/content/en/docs/reference/changelog.md:12]:
```suggestion
- Added support to EKS Anywhere for bare metal cluster provisioning with provider [tinkerbell](https://tinkerbell.org/).
```

**Review by chrisnegus — COMMENTED** [review]
Just a couple of minor suggestions.

File: docs/content/en/docs/reference/changelog.md (hunk 1 above)
Comment (pokearu) [reply docs/content/en/docs/reference/changelog.md:7]: I think we usually remove it when we back port to the release branch? But main retains it?

Comment (vivek-koppuru) [reply docs/content/en/docs/reference/changelog.md:7]: This is for the main branch (it acts as staging) so once we bring the commit to the release branch (which will be published), then it won't show. I think before we accidentally brought the unreleased section to the release branch.

Comment (chrisnegus) [reply docs/content/en/docs/reference/changelog.md:7]: Got it. Thanks.

**Pull request merged**